java-frontend/src/main/java/org/sonar/java/se/checks/XxeProcessingCheck.java

```diff
@@ -145,57 +152,45 @@
   private static final Map<MethodMatcher, Predicate<ConstraintsByDomain>> CONDITIONS_FOR_SECURED_BY_TYPE_NEW_CLASS = ImmutableMap.of(
     SAX_BUILDER_CONSTRUCTOR,
     c -> (c.hasConstraint(AttributeDTD.SECURED) && c.hasConstraint(AttributeSchema.SECURED))
-      || c.hasConstraint(FeatureDisallowDoctypeDecl.SECURED));
+      || c.hasConstraint(FeatureDisallowDoctypeDecl.SECURED),
+    SAX_READER_CONSTRUCTOR,
+    c -> c.hasConstraint(FeatureDisallowDoctypeDecl.SECURED)
+      || c.hasConstraint(FeatureExternalGeneralEntities.SECURED));
 
   private static final MethodMatcherCollection FEATURES_AND_PROPERTIES_SETTERS = MethodMatcherCollection.create(
-    MethodMatcher.create()
-      .typeDefinition(TypeCriteria.subtypeOf(XML_INPUT_FACTORY))
-      .name(SET_PROPERTY)
-      .parameters(JAVA_LANG_STRING, JAVA_LANG_OBJECT),
-    MethodMatcher.create()
-      .typeDefinition(TypeCriteria.subtypeOf(DOCUMENT_BUILDER_FACTORY))
-      .name(SET_FEATURE)
-      .parameters(JAVA_LANG_STRING, BOOLEAN),
+    setPropertyMatcher(XML_INPUT_FACTORY),
+    setFeatureMatcher(DOCUMENT_BUILDER_FACTORY),
     MethodMatcher.create()
```

Comment:
why not create as well a `setAttributeMatcher`  then? It seems to me that it would simplify the whole list

Reply:
True, I though it was used only once